packages/mui-material/src/OverridableComponentAugmentation.ts

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { DistributiveOmit } from '@mui/types';
+
+declare module '@mui/material/OverridableComponent' {
+  /**
+   * A component whose root component can be controlled via a `component` prop.
+   *
+   * Adjusts valid props based on the type of `component`.
+   */
+  interface OverridableComponent<M extends OverridableTypeMap> {
+    <C extends React.ElementType>(
+      props: {
+        /**
+         * The component used for the root node.
+         * Either a string to use a HTML element or a component.
+         */
+        component: C;
+      } & OverridePropsVer2<M, C>,
+    ): JSX.Element;
+    (props: DefaultComponentPropsVer2<M>): JSX.Element;
+  }
+
+  /**
+   * Props of the component if `component={Component}` is used.
+   */
+  // prettier-ignore
+  type OverridePropsVer2<
+    M extends OverridableTypeMap,
+    C extends React.ElementType,
```

Comment:
If we're going to use these types across the codebase at some point, it would be great to rename the generic props to something more meaningful.

Reply:
I am keeping it as it was before, so that it is visible what exactly are the changes between the two, but yes, we can rename the generics at any point in time :) 

Can be done in a dedicated PR where both the existing and the new types will be renamed.